cpmpy/solvers/pysat.py

```diff
@@ -320,49 +355,61 @@ def add(self, cpm_expr_orig):
 
             What 'supported' means depends on the solver capabilities, and in effect on what transformations
             are applied in `transform()`.
+        """
+        # add new user vars to the set
+        get_variables(cpm_expr_orig, collect=self.user_vars)
+
+        # transform and post the constraints
+        for cpm_expr in self.transform(cpm_expr_orig):
+            self._add_expr(cpm_expr)
 
-      """
-      # add new user vars to the set
-      get_variables(cpm_expr_orig, collect=self.user_vars)
+        return self
+
+    __add__ = add  # avoid redirect in superclass
 
-      # transform and post the constraints
-      for cpm_expr in self.transform(cpm_expr_orig):
+    def _add_clauses(self, clauses):
```

Comment:
I'm not sure why you create this indirection?
function calls are not free in Python...

if you want the code in `_add_expr()` look cleaner, then there define
add_clauses = self.pysat_solver.append_formula

and that will have the 'alias' effect

Reply:
where/how do you print the clauses? or was this for during development? in the latter, its no longer needed now...

Reply:
it would actually make the pull request cleaner, as more than half of the lines I see below are just from introducing this indirection...